File: problems/valid_anagram/solution_1/python/main.py
def isAnagram(s: str, t: str) -> bool:
    if len(s) != len(t):
        return False

    alphabet = [0] * 26
    
    for char in s:
        alphabet[ord(char) - ord("a")] += 1

    for char in t:
        index = ord(char) - ord("a")
        alphabet[index] -= 1

        if alphabet[index] < 0:
            return False

    return True

def testAnagram():
    testCases = [
        ("anagram", "nagaram", True),
        ("rat", "car", False),
    ]

    for s, t, expected in testCases:
        result = isAnagram(s, t)
        print(f"isAnagram({s}, {t}) = {result}, expected = {expected}")

    
testAnagram()

File: problems/valid_anagram/solution_1/golang/main.go
package main

import "fmt"

func isAnagram(s, t string) bool {
	if len(s) != len(t) {
		return false
	}

	alphabet := make([]int, 26)
	
	for i := 0; i < len(s); i++ {
		alphabet[s[i]-'a']++
		alphabet[t[i]-'a']--
	}

	for _, count := range alphabet {
		if count != 0 {
			return false
		}
	}
	
	return true
}

func testAnagram() {
	testCases := []struct {
		s        string
		t        string
		expected bool
	}{
		{s: "anagram", t: "nagaram", expected: true},
		{s: "rat", t: "car", expected: false},
	}

	for _, tc := range testCases {
		result := isAnagram(tc.s, tc.t)
		fmt.Printf(
			"isAnagram(%q, %q) = %v | Expected: %v\n",
			tc.s,
			tc.t,
			result,
			tc.expected,
		)
	}
}

func main() {
	testAnagram()
}